packages/compiler-cli/ngcc/src/main.ts

```diff
@@ -148,6 +149,20 @@ export function mainNgcc(
   // NOTE: Avoid eagerly instantiating anything that might not be used when running sync/async or in
   //       master/worker process.
   const fileSystem = getFileSystem();
+
+
+  // Bail out early if the work is already done.
+  const supportedPropertiesToConsider = ensureSupportedProperties(propertiesToConsider);
+  const absoluteTargetEntryPointPath =
+      targetEntryPointPath !== undefined ? resolve(basePath, targetEntryPointPath) : undefined;
+  if (absoluteTargetEntryPointPath !== undefined &&
+      hasProcessedTargetEntryPoint(
+          fileSystem, absoluteTargetEntryPointPath, supportedPropertiesToConsider,
+          compileAllFormats)) {
+    logger.debug('The target entry-point has already been processed');
+    return;
```

Comment:
`mainNgcc` is expected to return a promise if `async` is true (returning `undefined` could break someone using it programatically and expecting a promise - not that there's anyone doing this, but better to be safe :grin:):

```suggestion
    return async ? Promise.resolve() : undefined;
```